### DISCUSSION TITLE
Windows not recognizing usb midi device name (RP2040)

### DISCUSSION THREAD

**Discussion by fapplin** [earlephilhower/arduino-pico#1244] [Q&A]
I've posted this in the Adafruit forum, but I thought I might get an answer, here.

I used the following 2 lines in some code in my Raspberry Pi Pico (not real names). It is a MIDI USB device. I wanted to use a name for my device other than "Pico W".

```
USBDevice.setManufacturerDescriptor("Fubar Mfg                      ");
USBDevice.setProductDescriptor     ("FubarDevice                    ");
```

On my linux machine, if I bring up a MIDI application (Surge XT as an example) - I see FubarDevice listed as a MIDI input I can use. I can do the same on an Android device. But, if I plug my device into a Windows machine - MIDI applications don't even see the device. If I comment out those 2 lines of code, recompile, and place the .uf2 file on the Pico device then Windows will see the MIDI device as Pico W.

Does anyone have an idea of how I can get Windows to recognize my device name for my MIDI USB device?

Thanks,

Frank

**Comment by earlephilhower** [comment]
Just a shot in the dark, but is there something related to the Windows drivers?  Do you need to rerun Zadig or some other tool to make it recognize a "new" type of device ("new" as far as Windows is concerned)?

**Reply by fapplin** [reply]
OK. this is kind of strange. If I change

```
USBDevice.setManufacturerDescriptor("Fubar Mfg                        ");
```
to

```
USBDevice.setManufacturerDescriptor("Raspberry Pi Pico W             ");
```

and I go to the Sound, Video, and game controllers in the Device Manager. I see the device with the changed device name (FoobarDevice). If I look at the Properties and then Driver/Driver Details. It has the exact same driver files as it would if it showed up as the device name "Pico W", but the MIDI applications still don't see the device.

**Comment by fapplin** [comment]
Proper way to change the device name.

https://arduino.github.io/arduino-cli/0.31/platform-specification/#referencing-another-core-variant-or-tool
